src/utils/iterators.py

```diff
@@ -0,0 +1,9 @@
+from collections.abc import Iterator
+
+
+def get_first[T](it: Iterator[T]) -> T | None:
+    """Get the first item from an iterable, or None if it's empty."""
+    try:
+        return next(it)
+    except StopIteration:
+        return None
```

Comment:
Maybe `default` is a bad name, but something like this

```suggestion
def get_first[T, V](it: Iterator[T], default: V = None) -> T | V:
    """Get the first item from an iterable, or None if it's empty."""
    try:
        return next(it)
    except StopIteration:
        return default
```